File: solutions/0014. Longest Common Prefix/0014.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        if len(strs) == 1:
            return strs[0]

        ans = ""
        mini = len(strs[0])
        isMatch = True

        for i in range(1, len(strs)):
            mini = min(mini, len(strs[i]))

        for i in range(mini):
            c = strs[0][i]
            for j in range(1, len(strs)):
                if c != strs[j][i]:
                    isMatch = False
                    break
            if not isMatch:
                break
            ans += c

        return ans

File: solutions/0014. Longest Common Prefix/0014.java
class Solution {
  public String longestCommonPrefix(String[] strs) {
    if (strs.length == 0)
      return "";

    for (int i = 0; i < strs[0].length(); ++i)
      for (int j = 1; j < strs.length; ++j)
        if (i == strs[j].length() || strs[j].charAt(i) != strs[0].charAt(i))
          return strs[0].substring(0, i);

    return strs[0];
  }
}